src/sema/eval.rs

```diff
@@ -729,6 +729,150 @@ pub(crate) fn eval_constants_in_expression(
                 (None, true)
             }
         }
+        Expression::More {
+            loc,
+            left,
+            right,
+        } => {
+            let left = eval_constants_in_expression(left, diagnostics).0;
+            let right = eval_constants_in_expression(right, diagnostics).0;
+
+            if let (
+                Some(Expression::NumberLiteral { value: left, .. }),
+                Some(Expression::NumberLiteral { value: right, .. }),
+            ) = (&left, &right)
+            {
+                (
+                    Some(Expression::BoolLiteral { 
+                        loc: *loc,
+                        value: left > right
+                    }),
+                    true,
+                )
```

Comment:
Please don't optimize the code in sema, this should be done in codegen. 

Also this does not take signed-ness into account. Unsigned/signed compare are different.